gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/request/OAuth2Request.java

```diff
@@ -244,17 +246,20 @@ public void setConfirmationMethodX5S256(String confirmationMethodX5S256) {
         this.confirmationMethodX5S256 = confirmationMethodX5S256;
     }
 
-    public boolean shouldGenerateIDToken() {
+    public boolean shouldGenerateIDToken(User endUser) {
```

Comment:
**todo**: instead of providing a new paramter, may be we can use the isClientOnly method of this class ? So instead of testing `endUser == null` we can simply do `this.isClientOnly()`. wdyt ?